coreblocks/func_blocks/fu/fpu/lza.py

```diff
@@ -0,0 +1,86 @@
+from amaranth import *
+from transactron import TModule, Method, def_method
+from coreblocks.func_blocks.fu.fpu.fpu_common import FPUParams
+
+
+class LZAMethodLayout:
+    """LZA module layouts for methods
+    Parameters
+    ----------
+    fpu_params: FPUParams
+        FPU parameters
+    """
+
+    def __init__(self, *, fpu_params: FPUParams):
+        self.predict_in_layout = [
+            ("sig_a", fpu_params.sig_width),
+            ("sig_b", fpu_params.sig_width),
+            ("carry", 1),
+        ]
+        self.predict_out_layout = [
+            ("shift_amount", range(fpu_params.sig_width)),
+            ("is_zero", 1),
+        ]
+
+
+class LZAModule(Elaboratable):
+    """LZA module
+
+    Parameters
+    ----------
+    fpu_params: FPUParams
+        FPU rounding module parameters
+
+    Attributes
+    ----------
+    predict_request: Method
+        Transactional method for initiating leading zeros prediction.
+        Takes 'predict_in_layout' as argument
+        Returns shift amount as 'predict_out_layout'
+    """
+
+    def __init__(self, *, fpu_params: FPUParams):
+
+        self.lza_params = fpu_params
+        self.method_layouts = LZAMethodLayout(fpu_params=self.lza_params)
+        self.predict_request = Method(
+            i=self.method_layouts.predict_in_layout,
+            o=self.method_layouts.predict_out_layout,
+        )
+
+    def elaborate(self, platform):
+        m = TModule()
+
+        @def_method(m, self.predict_request)
+        def _(arg):
```

Comment:
```suggestion
        def _(sig_a, sig_b, carry):
```
"new" argument syntax is preferred for small number of arguments

Reply:
Done